cadc-vos-server/src/main/java/ca/nrc/cadc/vos/server/web/action/FilesAction.java

```diff
@@ -0,0 +1,195 @@
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.URLEncoder;
+
+import org.apache.log4j.Logger;
+
+import ca.nrc.cadc.auth.AuthMethod;
+import ca.nrc.cadc.auth.AuthenticationUtil;
+import ca.nrc.cadc.reg.Capabilities;
+import ca.nrc.cadc.reg.Capability;
+import ca.nrc.cadc.reg.Interface;
+import ca.nrc.cadc.reg.Standards;
+import ca.nrc.cadc.reg.client.RegistryClient;
+import ca.nrc.cadc.rest.InlineContentHandler;
+import ca.nrc.cadc.rest.RestAction;
+import ca.nrc.cadc.vos.DataNode;
+import ca.nrc.cadc.vos.Direction;
+import ca.nrc.cadc.vos.Node;
+import ca.nrc.cadc.vos.NodeNotFoundException;
+import ca.nrc.cadc.vos.VOS;
+import ca.nrc.cadc.vos.VOSURI;
+import ca.nrc.cadc.vos.server.NodePersistence;
+import ca.nrc.cadc.vos.server.PathResolver;
+import ca.nrc.cadc.vos.server.VOSpacePluginFactory;
+import ca.nrc.cadc.vos.server.auth.VOSpaceAuthorizer;
+import ca.nrc.cadc.vos.server.util.BeanUtil;
+
+public class FilesAction extends RestAction {
+    
+    protected static Logger log = Logger.getLogger(FilesAction.class);
+    
+    public FilesAction() {
+        super();
+    }
+
+    @Override
+    protected InlineContentHandler getInlineContentHandler() {
+        return null;
+    }
+
+    @Override
+    public void doAction() throws Exception {
+  
+        String path = syncInput.getPath();
+        log.debug("path: " + path);
+        if (path == null) {
+            sendError(404, "Not Found");
+            return;
+        }
+        
+        String vosURIPrefix = initParams.get(BeanUtil.IVOA_VOS_URI);
+        String nodeURI = vosURIPrefix + "/" + path;
+        VOSURI vosURI = new VOSURI(nodeURI);
+        
+        VOSpacePluginFactory pluginFactory = new VOSpacePluginFactory();
+        NodePersistence np = pluginFactory.createNodePersistence();
+        VOSpaceAuthorizer authorizer = new VOSpaceAuthorizer(true);
+        authorizer.setNodePersistence(np);
+        PathResolver resolver = new PathResolver(np, false);
+        
+        Node node = null;
+        try {
+            node = resolver.resolveWithReadPermissionCheck(vosURI, authorizer, true);
+        } catch (NodeNotFoundException e) {
+            sendError(404, "Not Found");
+            return;
+        }
+        
+        log.debug("node: " + node);
+        // only data nodes (link nodes resolved above)
+        if (!(node instanceof DataNode)) {
+            sendError(400, "Not a DataNode");
+            return;
+        }
+        
+        // create the synctrans url
+        RegistryClient regClient = new RegistryClient();
+        AuthMethod authMethod = AuthenticationUtil.getAuthMethod(AuthenticationUtil.getCurrentSubject());
+        String requestProtocol = syncInput.getProtocol();
+        
+        Capabilities caps = regClient.getCapabilities(vosURI.getServiceURI());
+        Capability cap = caps.findCapability(Standards.VOSPACE_SYNC_21);
+        Interface ifc = null;
+        for (Interface next : cap.getInterfaces()) {
```

Comment:
It is slightly different that the common findInterface will do because it is preferring the URL with the scheme with which the client used to connect.
If we (or others) using this code only have HTTPS URLs in their capabilities endpoints, then only HTTPS will work.